code/pagetypes/ForumHolder.php

```diff
@@ -252,11 +242,24 @@ function getNumTopics() {
 	 * @return int Returns the number of distinct authors
 	 */
 	public function getNumAuthors() {
-		return DB::query("
-			SELECT COUNT(DISTINCT \"Post\".\"AuthorID\") 
-			FROM \"Post\" 
-			JOIN \"" . ForumHolder::baseForumTable() . "\" AS \"ForumPage\" ON \"Post\".\"ForumID\"=\"ForumPage\".\"ID\"
-			AND \"ForumPage\".\"ParentID\" = '" . $this->ID . "'")->value(); 		
+		return $this->getForumStatistics()['Authors'];
+	}
+
+	/**
+	 * Tally the number of posts, threads and authors from the Forum of this ForumHolder
+	 *
+	 * @return array Returns an key => value pair of the stats for the forum.
+	 */
+	public function getForumStatistics() {
```

Comment:
@tractorcow good point. Are you recommending reimplementing SQL approach for each of the getNumPost, getNumTopic, getNumAuthors method and removing this getForumStatistics method?